BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c

```diff
@@ -1364,61 +1578,38 @@ PciEnableDevices (
 }
 
 /**
-  Program Pci resources according to BAR type
+  Program Pci resources
 
-  @param [in] RootBridge    A pointer which has Root Bridges in ChildList
-  @param [in] BarType       Base address register type
-  @param [in] BaseAddress   The resource base address to be allocated
+  @param [in] EnumPolicy    PciEnum Policy with root bridge mask to be scanned
+  @param [in] RootBridges   A pointer which has Root Bridges in ChildList
 
-  @retval Address           Max address allocated for BarType
  **/
-UINT64
+VOID
 PciProgramResources (
-  IN  PCI_IO_DEVICE     *RootBridge,
-  IN  PCI_BAR_TYPE       BarType,
-  IN  UINT64             BaseAddress
-  )
+  IN CONST  PCI_ENUM_POLICY_INFO  *EnumPolicy,
+  IN CONST  PCI_RES_ALLOC_TABLE   *ResAllocTable,
+  IN CONST  PCI_IO_DEVICE         *RootBridges
+)
 {
-  LIST_ENTRY                *CurrentLink;
-  PCI_IO_DEVICE             *Root;
-  UINT64                     Address;
-
-  if ((BarType == PciBarTypeUnknown) || (BarType > PciBarTypePMem64)) {
-    return BaseAddress;
-  }
-
-  Address = BaseAddress;
-  CurrentLink = RootBridge->ChildList.ForwardLink;
-  while ((CurrentLink != NULL) && (CurrentLink != &RootBridge->ChildList)) {
-    Root = PCI_IO_DEVICE_FROM_LINK (CurrentLink);
-    CalculateResource (Root, BarType);
-    Address = ALIGN(Address, Root->PciBar[BarType - 1].Alignment);
-    Root->PciBar[BarType - 1].BaseAddress = Address;
-    ProgramResource (Root, BarType);
-
-    // Base Address for next device
-    Address += Root->PciBar[BarType - 1].Length;
-    CurrentLink = CurrentLink->ForwardLink;
+  if (ResAllocTable->NumOfEntries == 1) {

```

Comment:
why do we have two flows here ?
Is it possible to unify the flow using PCiProgramResorucesByBridge ?

Reply:
This was just to keep the original allocation policy even with multiple bridges. Let me use PCiProgramResorucesByBridge only.